src/bin/vip-import-validate-files.js

```diff
@@ -0,0 +1,153 @@
+		 *
+		 * Recommended structure: `uploads/year/month` (Single sites)
+		 */
+		const nestedDirectories = await findNestedDirectories( filePath );
+
+		if ( nestedDirectories ) {
+			folderStructureValidation( nestedDirectories );
+		}
+
+		/**
+		 * Individual file validations
+		 *
+		 * - Media file extension/type validation
+		 * - Filename validation
+		 * - Intermediate image validation
+		 */
+		let files;
+
+		try {
+			files = await readDir( nestedDirectories );
+
+			if ( ! files || ! files.length || files.length <= 0 ) {
+				console.error( chalk.red( '✕ Error:' ), 'Media files directory cannot be empty' );
+			}
+		} catch ( error ) {
+			console.error( chalk.red( '✕ Error:' ), `Unable to read directory ${ filePath }: ${ error.message }` );
+		}
+
+		/**
+		 * Media file extension validation
+		 *
+		 * Ensure that prohibited media file types are not used
+		 */
+
+		// Collect invalid files for error logging
+		const errorFileTypes = [];
+		const errorFileNames = [];
+		const intImagesObject = {};
+
+		// Iterate through each file to isolate the extension name
+		for ( const file of files ) {
+			// Check if file is a directory
+			const stats = await stat( nestedDirectories + '/' + file );
+			const isFolder = stats.isDirectory();
+
+			const extension = path.extname( file ); // Extract the extension of the file
+			const ext = extension.substr( 1 ); // We only want the ext name minus the period (e.g- .jpg -> jpg)
+			const extLowerCase = ext.toLowerCase(); // Change any uppercase extensions to lowercase
+
+			// Check for any invalid file extensions
+			// Returns true if ext is invalid; false if valid
+			const invalidExtensions = acceptedExtensions.indexOf( extLowerCase ) < 0;
+
+			// Collect files that have no extension, have invalid extensions,
+			// or are directories for error logging
+			if ( ! extension || invalidExtensions || isFolder ) {
+				errorFileTypes.push( file );
+			}
+
+			/**
+			 * Filename validation
+			 *
+			 * Ensure that filenames don't contain prohibited characters
+			 */
+
+			// Collect files that have invalid file names for error logging
+			if ( isFileSanitized( file ) ) {
+				errorFileNames.push( file );
+			}
+
+			/**
+			 * Intermediate image validation
+			 *
+			 * Detect any intermediate images.
+			 *
+			 * Intermediate images are copies of images that are resized, so you may have multiples of the same image.
+			 * You can resize an image directly on VIP so intermediate images are not necessary.
+			 */
+			const original = doesImageHaveExistingSource( file, nestedDirectories );
+
+			// If an image is an intermediate image, populate key/value pairs of the original image and intermediate image(s)
+			if ( original ) {
+				if ( intImagesObject[ original ] ) {
+					// Key: original image, value: intermediate image(s)
+					intImagesObject[ original ] = intImagesObject[ original ] + ', ' + file;
```

Comment:
Consider using string interpolation to avoid usage of the `+` sign

```javascript
intImagesObject[ original ] = `${intImagesObject[ original ]}, ${file}`;
```

Reply:
Oops, changed in 3c376e7!